documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -194,28 +638,92 @@ The auto config properties for distributed caches are:
+
+Occasionally you might need to rebuild the Lucene index by reconstructing it from the data stored in the Cache. You need to rebuild the index if you change the definition of what is indexed on your types, or if you change for example some _Analyzer_ parameter, as Analyzers affect how the index is defined. Also, you might need to rebuild the index if you had it destroyed by some system administration mistake. To rebuild the index just get a reference to the MassIndexer and start it; beware if might take some time as it needs to reprocess all data in the grid!
+
+[source,java]
+----
+// Blocking execution
+SearchManager searchManager = Search.getSearchManager(cache);
+searchManager.getMassIndexer().start();
+
+// Non blocking execution
+CompletableFuture<Void> future = searchManager.getMassIndexer().startAsyc();
+
+----
+
+TIP: This is also available as a `start` JMX operation on the link:{javadocroot}/jmxComponents.html#MassIndexer[MassIndexer MBean]
+registered under the name `org.infinispan:type=Query,manager="{name-of-cache-manager}",cache="{name-of-cache}",component=MassIndexer`.
+
+===== Indexless [[query.indexless]]
+
+TODO
+
+===== Hybrid [[query.hybrid]]
 
-==== Lucene Directory
-Infinispan Query isn't aware of where you store the indexes, it just passes the configuration of which _Lucene Directory_ implementation you want to use to the Hibernate Search engine. There are several _Lucene Directory_ implementations bundled, and you can plug your own or add third party implementations: the Directory is the IO API for Lucene to store the indexes.
+TODO
 
-The most common _Lucene Directory_ implementations used with _Infinispan Query_ are:
+===== Mapping Entities
 
+Infinispan relies on Hibernate Search rich API in order to define fine grained configuration for indexing at entity level.
+This configuration includes which fields are annotated, which analyzers should be used, how to map nested objects and so on.
+Detailed documentation is available at link:https://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-mapping[the Hibernate Search manual].
 
-* Ram - stores the index in a local map to the node. This index can't be shared.
-* Filesystem - stores the index in a locally mounted filesystem. This could be a network shared FS, but sharing this way is generally not recommended.
-* Infinispan - stores the index in a different dedicated Infinispan cache. This cache can be configured as replicated or distributed, to share the index among nodes. See also the dedicated chapter on the Lucene Directory in this guide.
+====== @DocumentId
+Unlike Hibernate Search, using _@DocumentId_ to mark a field as identifier does not apply to Infinispan values; in Infinispan Query the identifier for all _@Indexed_ objects is the key used to store the value. You can still customize how the key is indexed using a combination of _@Transformable_ , custom types and custom _FieldBridge_ implementations.
```

Comment:
It doesn't feel like that we explained that "all this stuff" is something we call the "Infinispan Query" component.
There's a "Query" chapter. But I'm not sure it's clear here what we mean with making a comparison between "Hibernate Search" [unlike] "Infinispan Query". So far we just mentioned that Infinispan's _queries_ use Hibernate Search..